src/worker.cc

```diff
@@ -1233,12 +1089,93 @@ worker_t::result_t to_response(const std::string& data,
 
 #endif
 
-service_worker_t::service_worker_t() : interrupt(nullptr) {
+// TODO: when we want to use this in mjolnir too we can move this into a private header
+// this is a wrapper of a third party lib that provides a client for statsd integration
+// since metrics are important both for on- and offline processing we keep the impl here
+// running services can use it and also data ETL that lives in mjolnir can use it
+struct statsd_client_t : public Statsd::StatsdClient {
+  statsd_client_t(const boost::property_tree::ptree& conf)
+      : Statsd::StatsdClient(conf.get<std::string>("statsd.host", ""),
+                             conf.get<int>("statsd.port", 8125),
+                             conf.get<std::string>("statsd.prefix", ""),
+                             conf.get<uint64_t>("statsd.batch_size", 500),
+                             0) {
+    auto added_tags = conf.get_child_optional("statsd.tags");
+    if (added_tags) {
+      for (const auto& tag : *added_tags) {
+        tags.push_back(tag.second.data());
+      }
+    }
+  }
+  std::vector<std::string> tags;
```

Comment:
at the moment the only point of this pimpl class is to store the tags from config. in the future we could add more custom stuff as we need it for other logging purposes